src/lib.rs

```diff
@@ -52,8 +66,11 @@ impl From<u8> for Direction {
 }
 
 #[cfg(feature = "table-decoder")]
-impl From<u8> for Direction {
-    fn from(s: u8) -> Self {
+/// The useful values of `s` are:
+/// - 0x17 | 0x7E | 0xE8 | 0x81
+/// - 0x2B | 0xBD | 0xD4 | 0x42
+impl Phase for DefaultPhase {
+    fn direction(&mut self, s: u8) -> Direction {
```

Comment:
I don't see `self` actually used here, is `&mut self` used in the trait because it's useful for other implementations?

Reply:
It just allows the most flexible implementation of `Phase`, at no cost.